department-of-veterans-affairs/vets-api
# [88977] adds `sec_ids` to MPI profile

## Summary

- Adds new MPI Profile attribute `sec_ids`; this uses the same parsing as `sec_id` but takes the entire array if multiple `sec_id` values are sent by MPI
- Adds (non-blocking) validation for multiple `sec_id` values to
  - SiS authentication to keep it inline with SSOe
    - Includes `MULTIPLE_SEC_ID = '103'` error code - this is currently unused but necessary for the SiS ids validation code and would be used in the event that we decide to make multiple `sec_id` values an auth-blocking error.
  - ToU `SignUpServiceUpdaterJob`

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/88977

## Testing done

- finding a working mock MPI profile with multiple `sec_id` values was difficult; I ended up adding a second `sec_id` line to the MPI response
  - `sec_id` MPI extension: `^PN^200PROV^USDVA^A`
  - example user: `vets.gov.user+2@gmail.com`
  - MPI yml file: `vets-api-mockdata/mvi/profile_icn/1013174704V087163.yml` 
  - updated response:
  ```yml
    <subject1 typeCode="SBJ">
    <patient classCode="PAT">
      <id extension="1013174704V087163^NI^200M^USVHA^P" root="2.16.840.1.113883.4.349"/>
      <id extension="06806afe818d4d32b39691547d54cd50^PN^200VIDM^USDVA^A" root="2.16.840.1.113883.4.349"/>
      # original sec_id
      <id extension="1013174704^PN^200PROV^USDVA^A" root="2.16.840.1.113883.4.349"/>
      # duplicated sec_id
      <id extension="1234567890^PN^200PROV^USDVA^A" root="2.16.840.1.113883.4.349"/>
      <id extension="220655^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
      <id extension="c78a2c52-eda7-49b1-9a63-1cf8390e213b^PN^200VLGN^USDVA^A" root="2.16.840.1.113883.4.349"/>
      <id extension="0000001013174704V087163000000^PI^200ESR^USVHA^A" root="2.16.840.1.113883.4.349"/>
      <statusCode code="active"/>
  ```
- ensure that the user you have chosen does not have a saved `TermsOfUseAgreement`
- start vets-api (including Sidekiq) & vets-website, then perform an authentication with your chosen user. You should find the following the ToU SuS log in your vets-api console:
  - Rails -- [TermsOfUse][SignUpServiceUpdaterJob] Multiple sec_id values detected -- { :icn => "1013174704V087163" }

## Logs

## What areas of the site does it impact?
- Authentication & terms of use

## Acceptance criteria

- [x]  I added logs of the developed feature
